File: src/test/java/personal/alexgarland/helloworldmvc/controller/HelloWorldControllerTest.java
package personal.alexgarland.helloworldmvc.controller;

import static org.junit.Assert.assertEquals;

import java.util.Map;

import org.junit.Test;
import org.springframework.web.servlet.ModelAndView;

public class HelloWorldControllerTest {

	@Test
	public void returnsExpectedMessageModelViewWithGivenName() {
		
		HelloWorldController controller = new HelloWorldController();
		
		ModelAndView result = controller.showMessage("Robert");
		Map<String, Object> model = result.getModel();
		
		assertEquals("helloworld", result.getViewName());
		
		assertEquals(2, model.size());
		assertEquals("Welcome to Spring MVC", model.get("message"));
		assertEquals("Robert", model.get("name"));
		
	}

}

File: src/test/kotlin/personal/alexgarland/helloworldmvc/controller/HelloWorldControllerTest.kt
package personal.alexgarland.helloworldmvc.controller

import org.junit.Assert.assertEquals
import org.junit.Test

class HelloWorldControllerTest {

    @Test
    fun returnsExpectedMessageModelViewWithGivenName() {

        val controller = HelloWorldController()

        val result = controller.showMessage("Robert")

        assertEquals("helloworld", result.viewName)

        val model = result.model

        assertEquals(2, model.size.toLong())
        assertEquals("Welcome to Spring MVC", model["message"])
        assertEquals("Robert", model["name"])

    }

}
